src/main.cpp

```diff
@@ -609,15 +671,24 @@ void loop()
     reset_config_due = false;
     resetConfig();
   }
+
+  static uint32_t pollTemperature_interval = 0;
+  if (TimeReached(pollTemperature_interval))
```

Comment:
Its possible that we read the temperature immediately twice (here and line #619). No deal breaker, but we could add a "last_temperature_read" timestamp and assess the freshness at line #619 and potentially skip the re-read. But I'm not sure whether this is worth the effort. Especially because Bricks tend to be turned on for a very long time so we'd have to make sure overflows for the timestamp are not catastrophic.

Reply:
Do you mean #691?
I moved sensors.requestTemperatures / sensors.getTempC out of printTemperature() into pollTemperature_interval. There is only this one place where the sensor is read.

Reply:
sorry :) now I see, in that case:

`Serial.println("##### MAIN: reading temperatureprintTemperature");` should be moved from line 690 to 680 and line 690 should become "printing temperature" or something similar. Did you experiment with a 1000ms temperature read interval? IMHO would also be OK.

Reply:
I see.
Originally i didn't want to clutter the serial monitor with those temperature values. Thats why I left printTemperature in the main_interval.
You can also do a 1s read out interval, no problem.
But the main problem stays: the very "blocky" contactBackend() function. This stalls everything in this thread for several seconds.